apps/admin-page/src/app/pages/dashboard/dashboard.component.ts

```diff
@@ -1,10 +1,454 @@
+    } else {
+      return 0;
+    }
+  }
+
+  subtractShippingCost(shippingMethod: ShippingMethodEnum, orderTotal: number): number {
+    if (shippingMethod === ShippingMethodEnum.homeDelivery && orderTotal > 350) {
+      return 25;
+    } else if (shippingMethod === ShippingMethodEnum.homeDelivery && orderTotal <= 350) {
+      return 65;
+    } else if (shippingMethod === ShippingMethodEnum.pickUpPoint && orderTotal <= 350) {
+      return 45;
+    } else {
+      return 0;
+    }
+  }
+
+  getPeriodOrderCount(period: number) {
+    const thisPeriodOrderCount = this.getPeriodOrders(period).length;
+    if (period === 14) {
+      return (this.twoWeekOrders = thisPeriodOrderCount);
+    } else if (period === 30) {
+      return (this.monthOrders = thisPeriodOrderCount);
+    } else if (period === 90) {
+      return (this.threeMonthOrders = thisPeriodOrderCount);
+    } else {
+      return (this.sixMonthOrders = thisPeriodOrderCount);
+    }
+  }
+
+  calculatePeriodOrderCountDifference(periodStart: number, periodEnd: number) {
+    const thisPeriodOrderCount = this.getPeriodOrders(periodStart).length;
+    const lastPeriodOrderCount = this.getPreviousPeriodOrders(periodStart, periodEnd).length;
+    if (periodStart === 14 && periodEnd === 28) {
+      this.twoPastWeekOrders = lastPeriodOrderCount;
+    } else if (periodStart === 30 && periodEnd === 60) {
+      this.monthPastOrders = lastPeriodOrderCount;
+    } else if (periodStart === 90 && periodEnd === 180) {
+      this.threePastMonthOrders = lastPeriodOrderCount;
+    } else {
+      this.sixPastMonthOrders = lastPeriodOrderCount;
+    }
+    return this.getPercentDiff(thisPeriodOrderCount, lastPeriodOrderCount);
+  }
+
+  getPeriodRevenue(period: number) {
+    const thisPeriodOrders = this.getPeriodOrders(period);
+    let thisPeriodRevenue = 0;
+    if (period === 14) {
+      thisPeriodOrders.forEach((order) => {
+        thisPeriodRevenue += order.total;
+      });
+      this.twoWeekRevenue = thisPeriodRevenue;
+    } else if (period === 30) {
+      thisPeriodOrders.forEach((order) => {
+        thisPeriodRevenue += order.total;
+      });
+      this.monthRevenue = thisPeriodRevenue;
+    } else if (period === 90) {
+      thisPeriodOrders.forEach((order) => {
+        thisPeriodRevenue += order.total;
+      });
+      this.threeMonthRevenue = thisPeriodRevenue;
+    } else {
+      thisPeriodOrders.forEach((order) => {
+        thisPeriodRevenue += order.total;
+      });
+      this.sixMonthRevenue = thisPeriodRevenue;
+    }
+    return thisPeriodRevenue;
+  }
+
+  calculatePeriodRevenueDifference(periodStart: number, periodEnd: number) {
+    const thisPeriodOrders = this.getPeriodOrders(periodStart);
+    const lastPeriodOrders = this.getPreviousPeriodOrders(periodStart, periodEnd);
+    let thisPeriodRevenue = 0;
+    let lastPeriodRevenue = 0;
+
+    if (periodStart === 14 && periodEnd === 28) {
+      thisPeriodOrders.forEach((order) => {
```

Comment:
Same applies here. Take the thisPeriodOrders.forEach out of the if and have it before so that code is only written once. It occurs in all cases anyways.